### ISSUE TITLE
Llava with LoRA experiments

### ISSUE THREAD

**Issue by gorjanradevski** [TRI-ML/prismatic-vlms#16]
Great work! One quick question: in the paper you've reproduced the results from Llava. Additionally, for the Prismatic models experiments, you are fine-tuning the whole LM. I'm wondering did you try using LoRA in the LM for the Llava experiments as well as for the Prismatic models experiments? More specifically, why did you opt for the fine-tuning the entire LM instead of training LoRA modules?

Thanks!

**Comment by NMS05** [comment]
Hi @gorjanradevski,

The weights and training logs are released by the authors in huggingface (https://huggingface.co/TRI-ML/prismatic-vlms/tree/main)

As for LoRA, I tried a simple implementation with SigLIP + Phi-2 + LoRA here (https://github.com/NMS05/Prismatic-SigLIP-Phi2-LoRA-VLM).

Thanks,

**Comment by zeyuanyin** [comment]
Hi, @NMS05. Thanks for your great implementation, which helps me a lot.

**Comment by siddk** [comment]
Awesome implementation @NMS05! Would love to add first-class LoRA support to this repo - would you be willing to contribute a PR?

**Comment by zeyuanyin** [comment]
Hi @NMS05, I would like to know if you have evaluated the finetuned model. I download and evaluate `my_phi2_siglip_model` via https://github.com/TRI-ML/vlm-evaluation. But the results are pretty bad. Or is there something wrong in eval repo? @siddk. Could you please double-check https://github.com/NMS05/Prismatic-SigLIP-Phi2-LoRA-VLM#model-weights and share your evaluation results? Thanks a lot.

|                             	| vqa-v2 	|   gqa 	| vizwiz 	| text-vqa-ocr 	| text-vqa 	|
|----------------------------:	|-------:	|------:	|-------:	|-------------:	|---------:	|
| phi2_siglip_lora (slim eval set) 	|  6.950 	| 3.520 	|  0.290 	|       0.283 	|    0.068 	|

**Comment by siddk** [comment]
Hmm so the evaluation code was able to fully reproduce the paper results for all existing models (as of last week); not sure if there's something different that needs to be changed for Phi-2 models (e.g., tokenization/generation parameters).

I'll try to take a look when I can (or might just try training a Phi-2 model). But would be great if @NMS05 could weigh in on their results!

**Comment by NMS05** [comment]
Hi @zeyuanyin, 

After I trained the SigLIP-Phi2(LoRA) model, I only did qualitative analysis using this [notebook](https://github.com/NMS05/Prismatic-SigLIP-Phi2-LoRA-VLM/blob/main/scripts/inference.ipynb). Despite training on a large dataset, I observed that object hallucinations were presistent both for VQA and captioning tasks. Not sure where things went wrong, since the "finetune" loss curve looks smooth. Will try to improve things in the next iteration.

I initially evaluated the finetuned model using [prismatic-eval](https://github.com/TRI-ML/vlm-evaluation) repo for vqa-v2-slim. For a vast majority of the questions the model doesnot produce any answer leading to very low performance. Not sure if there is a mismatch between prompts for train and eval. 
![image](https://github.com/TRI-ML/prismatic-vlms/assets/96108723/0b6af641-28ef-4083-b8d0-249f3ba4c110)

Thanks,

**Comment by arashakb** [comment]
So basically @NMS05 made the results better by doing two stage fine tuning?  I don't understand how the results got better.
